lib/device.c

```diff
@@ -387,6 +387,34 @@ int sync_save_tracks(const struct sync_device *d)
 
 #ifndef SYNC_PLAYER
 
+static int send_set_key_cmd(const struct sync_device *d, const struct sync_track *t, int row, float val, enum key_type type)
+{
+	unsigned char cmd = SET_KEY, _type = type;
+	uint32_t _track, _row;
+	union {
+		float f;
+		uint32_t i;
+	} v;
+
+	v.f = val;
+	v.i = htonl(v.i);
```

Comment:
I wonder if this should be factored out into a helper, as it's used somewhere else aswell... something like this?

```c
static uint32_t encode_float(float val)
{
	union {
		float f;
		uint32_t i;
	} u;
	u.f = val;
	return htonl(v.i);
}
```

Reply:
sounds good

Reply:
Right, thanks for pointing that out! :)